### DISCUSSION TITLE
Anbindung 5573-1

### DISCUSSION THREAD

**Discussion by falkio8** [Tom-Bom-badil/samson_trovis_557x#13] [General (de/en)]
Moin,

wie vermutlich alle hier bin ich über die Suche nach eine Alternativen zu den Samson Schnittstellen (kaum zu bekommen) auf diese Seite gestossen. Ich habe eine 5573-1 die ich demnächst einbauen werden und diese möchte ich hautpsächlich mit Trovis View auslesen und parametrieren können. Wenn das Ganze noch über LAN läuft umso besser.

Ich habe bereits in diesem GitHub alles durchgelesen, ich habe mir die Einträge in dem Microcontroller Forum (https://www.mikrocontroller.net/topic/346223?page=single) durchgelesen. Habe also mittlerweile einen halbwegs guten Überblick was zu tun ist. Für mich stellt sich jetzt die Frage ob ich es mit einem USR-TCP232-E2 probiere (etwas mehr Bastelarbeit) oder wie einer im Microcontroller.net schrieb mit einem USR-TCP232-410s (https://www.pusr.com/products/modbus-serial-to-ethernet-converters-usr-tcp232-410s.html). Das Teil wird etwas solider und ist ggf. etwas einfacher mit Strom zu versorgen?
Gibt es hierzu ein paar Anregungen? Müsste hier dann der RS232 Anschluss genommen werden oder der RS485?

Am Ende soll die 5573 im LAN erreichbar sein für TrovisView.

Danke

**Comment by Tom-Bom-badil** [comment]
Die in diesem Projekt verwendete Schnittstelle ist elektrisch RS232. Der Adapter ist Geschmackssache, mittlerweile findet man hier im Wiki und in den Kommentaren eine große Auswahl. Hinweis: Die Versorgungsspannung immer nur von einer Seite (Trovis _oder_ Netzteil) einspeisen.

/tom

**Reply by falkio8** [reply]
Hi, ich verzweifle hier gerade. Ich weiss die 5573-1 ist nicht dein Spezialgebiet aber vielleicht übersehe ich was:

5573-1004

Ich habe mir hier auf deine Seite als auch auf den Quellen mikrocontroller.net und knx Forum alles durchgelesen was es zu dem Thema gibt. War der Meinung ich bin gut vorbereitet. Was habe ich bisher gemacht:

In der 5573-1 Modbus aktiviert, Modem deaktiviert.
Das USR-TCP232-410s entsprechend deiner Anleitung konfiguriert. Ein RJ45 auf RS232 Kabel konfektioniert nach den PIN Belegungen die ich hier gefunden habe: 1 -> TxD, 8 -> RxD, 7 -> GND (habe auch testhalber PIN 1 und 8 schon getauscht.
Hinweis: zwischen 7 und 8 messe ich -6V. Zwischen 1 und 7 nichts.

Versuche mit der 55Pro App und Trovis View 4 sind alle am Kommunikationsproblem gescheitert. Wenn ich einen Versuch initiiere blinkt die 232Tx Lampe am USR kurz rot und das wars. Gibt es noch eine etwas aussagekräftigere Möglichkeit zu gucken ob man überhaupt kommunizieren kann und die Pins die richtigen sind? Habe ja nirgendwo sowas wie ein log window.

edit: kann es sein das die 5573-1 nur TTL und nicht RS232 kann? Manche Nutzer im MC Forum haben einen RS232 zu TTL Adapter zwischengeschaltet. In der Anleitung der 5573-1 steht auch nichts von RS232 sondern nur TTL. Und beim mein PUSR 410s steht auch nichts von TTL. Ich verstehe zu wenig davon aber sind dann hier die Pegel evtl. falsch? Habe ich den falschen Adapter?

Hier Bilder von allem was ich eingestellt habe:

![image](https://github.com/user-attachments/assets/3c4b2638-ace6-4026-af82-10fd29c2e099)
![Screenshot 2024-08-31 115226](https://github.com/user-attachments/assets/79592d63-1ea4-4286-995d-54069aa2bbec)
![Screenshot 2024-08-31 120856](https://github.com/user-attachments/assets/2b6e872d-787d-4574-99c8-c7f233ff7a9b)
![Screenshot 2024-08-31 114948](https://github.com/user-attachments/assets/0f3caffe-af6f-4265-a86f-54c2c9e58013)
![Screenshot 2024-08-31 115153](https://github.com/user-attachments/assets/f4959266-1b7c-401b-8769-207206b02967)
![scs01](https://github.com/user-attachments/assets/526a456f-10fb-440a-bf79-bd6829193144)
![scs02](https://github.com/user-attachments/assets/5d422812-ed44-4f98-84b3-c229b4815ba7)

**Comment by Tom-Bom-badil** [comment]
> edit: kann es sein das die 5573-1 nur TTL und nicht RS232 kann?

3.3V TTL liegt erstmal grundsätzlich innerhalb der definierten Signalpegel von RS232 (3...25V). Allerdings war es tatsächlich in Einzelfällen schon notwendig, einen Pegelwandler dazwischenzuschalten; kürzlich erst [hier](https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/14) (aber da war die Ausgangssituation eine andere).

Ich würde erstmal auf die Verkabelung bzw. deren Pinout tippen. Wie man [hier](https://github.com/Tom-Bom-badil/samson_trovis_557x/wiki/Modellspezifische-Informationen#5573) sieht, scheint es je nach (Unter-)Modell unterschiedliche Belegungen zu geben. Die 5573 war mir diesbezüglich schon immer ein kleines Mysterium, da unterschiedliche Verkabelungen kursieren.

Ich bin aber auch der Meinung, dass die 232RX und 232TX LED's bei korrekter Verkabelung zumindest ab und zu aktiv sein müssten. Allerdings kenne ich den von Dir verwendeten 410s nicht; hab hier zwar ein ähnliches aussehendes Modell auf RS485-Basis an meiner KWL, aber das ist eben ein ganz anderes Bussystem. Notfalls halt mal für 'nen Euro den im ersten Link genannten bidirektionalen Pegelwandler ordern ...

/tom

**Reply by falkio8** [reply]
Also ich habe es mit einem TTL Adapter und den Pins 1, 2 und 8 geschafft mit Tovis View und mit 55Pro eine Verbindung herzustellen und ich kann die Wert auslesen und mir im Trendviewer alles angucken. Was ich nicht verstehe ist, wenn ich zb. im 55Pro versuche die Parameter auszulesen bekomme ich ein EEPROM Fehler. Wenn ich im View versuche einen Parameter zu ändern bekomme ich die Fehlermeldung "Schreibschutzfehler". Ich habe aber in der Konfiguration die Schlüsselzahl 1732 angegeben.

Und wie kommt man auf die Benutzerebene Experte statt Kunde?

![grafik](https://github.com/user-attachments/assets/2b691c6a-691f-42d3-b9f6-d1c24e10c9c0)

![IMG_2249](https://github.com/user-attachments/assets/1d9c8daf-a11b-4689-81ce-ccecf496ba12)

**Comment by Tom-Bom-badil** [comment]
Zumindest die zweite Frage kann ich beantworten: Gemäß [diesem Handbuch](https://rvb.co.at/wp-content/uploads/2017/05/Samson-Vetilstellmotor-5824-und-5825.pdf#page=38) ist '_samson_' das passende Kennwort.

Zu den Fehlern kann ich aus der Ferne nur raten:
- Welche Firmware hast Du auf dem Regler?
- Passt TrovisView zu dieser Firmware, und wird beim Start von TV der richtige Regler mit dem richtigen Firmwarestand ausgewählt?
- Ist die Schlüsselzahl tatsächlich 1732, oder wurde diese geändert? (mal am Regler ausprobieren)
- zu 55Pro: Hier gibt es nach meinem Eindruck seit Jahren keine Weiterentwicklung, evtl ist die Firmware zu neu.

/tom

**Reply by falkio8** [reply]
Danke. Das Passwort habe ich auch gefunden aber nach Eingabe kann ich nicht auf ok drücken.
FW ist 2.63 und beim Start von View habe ich auch die Range angewählt 2.60-2.69.
Schlüsselzahl ist 1732. Damit kann ich auch an der TROVIS selbst alles einstellen.

Irgendwie habe ich offenbar keinen Schreibzugriff per TCP. An der Kommunikation selbst kann es ja nicht liegen oder? Da ich mich ja verbinden kann und alles auslesen kann.

**Comment by Tom-Bom-badil** [comment]
Ich kenne fehlerhafte Schreibvorgänge eigentlich nur vom Samson Firmware-Updater. Updates kann ich nur einspielen, wenn ich statt ModbusTCP per [VCom](https://www.pusr.com/support/download/usr-vcom-setup-software.html) einen virtuellen seriellen Port COMx: erstelle und den in TrovisView verwende.

Bei Updates per TCP wird die Firmware zwar initial geflashed, aber da meine Adapter grundsätzlich direkt von der Trovis gespeist werden, werden diese mit dem Reglerneustart 'kaltgestartet'. Und der Bootvorgang samt anschließender TCP-Verbindungswiederherstellung scheint dem Updater zu lange zu dauern, so dass das Update mit timeout fehlschlägt.

Ist also ein anderes Fehlerverhalten als bei Dir. Die grundsätzliche Kommunikation (also Abfrage an Regler - Antwort vom Regler) scheint ja nun zu laufen, sonst würdest Du in TrovisView keine Werte sehen. Bin ein wenig ratlos - welchen TTL-Adapter verwendest Du, und wird der von der Trovis oder direkt gespeist?

/tom

**Reply by falkio8** [reply]
Ich konnte mittlerweile auch ein Firmwareupdate erfolgreich aufspielen auf die 3.00.04. Mein TTL Adapter ist von Digitus und benötigt keine zusätzliche Spannungsversorgung. EIn Bild davon ist in deinem Sammelpost für erfolgreiche Setups.

Ich habe immer versucht im online Modus Parameter zu ändern. Ich habe das Gefühl, dass die Schnittstelle dazu nicht in der Lage ist. Offline kann ich die Parameter runterladen. Nur ein hochladen habe ich noch nicht getestet. Der laufdende Trendviewer scheint die Schnittstelle schon komplett auszulasten. Ab und zu bekomme ich dort auch Kommunikationsfehler. Und ein Durchklicken der Parameterordner dauert recht lange.
Vielleicht erwarte ich hier auch nur zu viel.

**Reply by Tom-Bom-badil** [reply]
So mache ich es hier: Alles in TrovisView herunterladen, offline lokal die gewünschten Parameter ändern, dann alles in einem Rutsch wieder hochladen. Allerdings lasse ich nebenbei auch keine Plots generieren oder Datensammlungen mitlaufen, das erledigt eh die Hausautomation für mich. Vielleicht liegt genau da Dein Problem - Überlastung auf dem Bus durch 'online' Dauerpolling diverser Parameter.

/tom
